httpx/_exceptions.py

```diff
@@ -235,7 +236,8 @@ def map_exceptions(
         if mapped_exc is None:
             raise
 
-        raise mapped_exc(exc) from None
+        message = str(exc)
+        raise mapped_exc(message, **kwargs) from None  # type: ignore
```

Comment:
We have to `type: ignore` here, because `**kwargs` isn't *generally* appropriate to any exception class, so it's up to us to ensure that calling code uses it appropriately with whichever exceptions are being mapped too.